### PR TITLE
Create repository from template

### PR DESCRIPTION
**Pull request by isouza-daitan**
Adds a new `create_repo_from_template` method to `AuthenticatedUser` and `Organization` that uses the new `generate` [endpoint](https://developer.github.com/v3/repos/#create-repository-using-a-repository-template) of a repo.

I also added a new `is_template` attribute to the `Repository` class. This attribute will only be sent by Github when we use the feature's Accept header which I only added to the `POST` to create the new repo.

The documentation mentions a `template_repository` attribute but Github never sends it.

### CODE REVIEW

File: github/AuthenticatedUser.py [hunk 1]
```diff
@@ -498,6 +498,47 @@ def create_fork(self, repo):
...
+    ):
+        """
+        :calls: `POST  /repos/:template_owner/:template_repo/generate <https://developer.github.com/v3/repos/#create-repository-using-a-repository-template>`
+        :param name: string
+        :param repo :class:`github.Repository.Repository`
+        :param description: string
+        :param private: bool
+        :rtype: :class:`github.Repository.Repository`
+        """
+        assert isinstance(repo, github.Repository.Repository), repo
```
Comment (andyfeller) [inline github/AuthenticatedUser.py]: Should this be:
```suggestion
        assert isinstance(name, str), name
```

Comment (isouza-daitan) [reply github/AuthenticatedUser.py]: My first implementation used strings but I changed it to `Repository` to match the pattern in `create_fork`, which is a very similar method.

Comment (andyfeller) [reply github/AuthenticatedUser.py]: ah well this line looks like a duplicate for the line beneath it

Comment (isouza-daitan) [reply github/AuthenticatedUser.py]: Thanks for catching this!

**Pull request merged**